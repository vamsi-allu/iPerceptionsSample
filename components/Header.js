import React from 'react';
import {StyleSheet, View, Text, Platform, Dimensions} from 'react-native';

const Header = (props) => {
  return (
    <View style={styles.header}>
      <Text allowFontScaling={false} style={styles.title}>
        {props.title}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    width: '100%',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    borderBottomColor: 'black',
    borderBottomWidth: 1,
    backgroundColor: '#3b5998',
    ...Platform.select({
      ios: {
        height: 100,
        paddingTop: Dimensions.get('window').width > 400 ? 50 : 25,
        paddingBottom: 20,
      },
      android: {
        height: 60,
        paddingBottom: 10,
      },
    }),
  },
  title: {
    fontSize: 20,
    color: 'white',
    fontWeight: '600',
  },
});

export default Header;
