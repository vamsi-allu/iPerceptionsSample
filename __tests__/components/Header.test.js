import 'jsdom-global/register';
import React from 'react';
import Header from '../../components/Header';
import Enzyme, {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

describe('Header Component', () => {
  it('renders correctly enzyme', () => {
    const props = {title: 'sampleTitle'};
    const wrapper = shallow(<Header {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('check the title is loaded properly', () => {
    const props = {title: 'sampleTitle'};
    const component = shallow(<Header {...props} />);

    expect(component.find('Text')).toHaveLength(1);
  });
});
