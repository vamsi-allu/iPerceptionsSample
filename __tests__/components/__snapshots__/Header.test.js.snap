// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly enzyme 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3b5998",
      "borderBottomColor": "black",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 100,
      "justifyContent": "center",
      "paddingBottom": 20,
      "paddingTop": 50,
      "width": "100%",
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "600",
      }
    }
  >
    sampleTitle
  </Text>
</View>
`;
